### ISSUE TITLE
Upgrade libpng to 1.6.37 (fixes vulnerability)

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#30564]
<div id="comment:0"></div>

The [libpng homepage](http://www.libpng.org/pub/png/libpng.html) warns:

> Vulnerability Warning
>
>
> libpng versions 1.6.36 and earlier have a use-after-free bug in
> the simplified libpng API png_image_free(). It has been assigned
> ID [CVE-2019-7317](http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-7317).
> The vulnerability is fixed in version 1.6.37,
> released on 15 April 2019.

Before this ticket we have libpng 1.6.29 which
has the vulnerability. This ticket upgrades to
libpng 1.6.37 which fixes it.

Previous update: #22159 (1.6.29)

Tarball: see checksums.ini

CC:  @jpflori @frederichan-IMJPRG @tscrim @slel @dimpase

Component: **packages: standard**

Author: **Matthias Koeppe**

Reviewer: **Dima Pasechnik**

_Issue created by migration from https://trac.sagemath.org/ticket/30564_

**Comment by slel** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,4 +1,17 @@
-http://www.libpng.org/pub/png/libpng.html
+The [libpng homepage](http://www.libpng.org/pub/png/libpng.html) warns:
+
+> Vulnerability Warning
+>
+>
+> libpng versions 1.6.36 and earlier have a use-after-free bug in
+> the simplified libpng API png_image_free(). It has been assigned
+> ID [CVE-2019-7317](http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-7317).
+> The vulnerability is fixed in version 1.6.37,
+> released on 15 April 2019.
+
+Before this ticket we have libpng 1.6.29 which
+has the vulnerability. This ticket upgrades to
+libpng 1.6.37 which fixes it.
 
 Previous update: #22159 (1.6.29)
 
``````

**Comment by mkoeppe** [comment]
Removed branch from ticket description; replaced by PR #38522.
